parachain/pallets/bridge/src/primitives.rs

```diff
@@ -0,0 +1,27 @@
+use frame_support::dispatch::{Parameter, DispatchResult};
+use sp_runtime::RuntimeDebug;
+use codec::{Encode, Decode};
+use artemis_core::Message;
+
+use sp_std::fmt::Debug;
+use sp_runtime::traits::{Member, MaybeSerializeDeserialize};
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, Default, RuntimeDebug)]
+pub struct InboundChannelData {
+	pub nonce: u64
+}
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, Default, RuntimeDebug)]
+pub struct OutboundChannelData {
+	pub nonce: u64
+}
+
+pub trait InboundChannel<AccountId>
+where
+	AccountId: Parameter + Member + Ord + MaybeSerializeDeserialize + Debug
+{
+	fn submit(&mut self, relayer: &AccountId, message: &Message) -> DispatchResult;
```

Comment:
I landed #203 which adds `submit_bulk` to the bridge and the verifier. I think it's fine to call `channel.submit` one by one after verification though... unless it's a real perf bottleneck. Am I correct it saying there will be 10 writes to storage for a batch of 10 messages? Regardless, premature optimization and all that jazz

Reply:
Am gonna leave `submit_bulk` alone for now. Can always move it into a channel in the next PR.